### ISSUE TITLE
empty file issue

### ISSUE THREAD

**Issue by koprda** [richardgirges/express-fileupload#226]
uploading "empty" zero-length file end with error in both modes memory and temp file

simple reproduction - truncate test file basketball.png and run test

**Comment by RomanBurunkov** [comment]
That is strange.
Can you turn module's debug output on and share it.

**Comment by RomanBurunkov** [comment]
@koprda 

I'm not able to reproduce that.
What version and system do u have?

See output from my test:

```
Express server listening on port  8000
Express-file-upload: Temporary file path is /projects/express-fileupload-project/temp/tmp-1-1593153213703
Express-file-upload: New upload started sampleFile->emptyfile.txt, bytes:0
Express-file-upload: Temporary file path is /projects/express-fileupload-project/temp/tmp-3-1593153213704
Express-file-upload: New upload started sampleFile1->tmp-2-1593153213704, bytes:0
Express-file-upload: Temporary file path is /projects/express-fileupload-project/temp/tmp-5-1593153213705
Express-file-upload: New upload started sampleFile2->tmp-4-1593153213705, bytes:0
Express-file-upload: Upload finished sampleFile->emptyfile.txt, bytes:0
Express-file-upload: Upload sampleFile->emptyfile.txt completed, bytes:0.
Express-file-upload: Upload finished sampleFile1->tmp-2-1593153213704, bytes:0
Express-file-upload: Upload finished sampleFile2->tmp-4-1593153213705, bytes:0
req.files >>> {
  sampleFile: {
    name: 'emptyfile.txt',
    data: <Buffer >,
    size: 0,
    encoding: '7bit',
    tempFilePath: '/projects/express-fileupload-project/temp/tmp-1-1593153213703',
    truncated: false,
    mimetype: 'text/plain',
    md5: 'd41d8cd98f00b204e9800998ecf8427e',
    mv: [Function: mv]
  }
}
Express-file-upload: Moving temporary file /projects/express-fileupload-project/temp/tmp-1-1593153213703 to /projects/express-fileupload-project/uploads/emptyfile.txt


```

**Comment by RomanBurunkov** [comment]
I was able to reproduce it.

Looking into this.

**Comment by RomanBurunkov** [comment]
In case of in-memory upload this PR #15  affects empty file upload.
@r3wt , could you please elaborate #14  issue?

Since we already have the following check in processMultipart:

```
      // Do not add file instance to the req.files if original name and size are empty.
      // Empty name and zero size indicates empty file field in the posted form.
      if (!name && size === 0) return;
```

Do we still need that check in fileFactory:

```
  // see: https://github.com/richardgirges/express-fileupload/issues/14
  // firefox uploads empty file in case of cache miss when f5ing page.
  // resulting in unexpected behavior. if there is no file data, the file is invalid.
  if (!fileUploadOptions.useTempFiles && !options.buffer.length) return;
```

**Comment by r3wt** [comment]
@RomanBurunkov Maybe. I'm not actually sure, we might have to summon Richard to see if he knows. This was so long ago I barely remember it.

**Comment by RomanBurunkov** [comment]
Hi @richardgirges ,  do you remember anything about issue #14 and PR #15. 
I can't figure out steps how to could be reproduced.

**Comment by RomanBurunkov** [comment]
Fixed with PR #233

**Comment by nusu** [comment]
I tried it, this doesn't seem fixed I'm going to fix it and push a following PR
